dvc/output/s3.py

```diff
@@ -35,6 +35,9 @@ def changed(self):
 
         return False
 
+    def changed_cache(self):
+        return self.project.cache.s3.changed_cache(self.md5)
```

Comment:
s3 doesn't have self.md5 property. We should add tests for all output types, you are currently only testing local one.

Reply:
How should I test it, @efiop , during the `test_data_cloud.py` is ok?

Reply:
GS and S3 require secure env vars, which are not available for pull requests outside of iterative. There is a lot of common logic with all the other remote types that are tested(hdfs, ssh, local, http), so as long as those tests pass we should be pretty safe in this PR. Travis will test it thoughtfully after it is merged. We should definitely start using something like an azurite for s3 and gs eventually, but it will do for now.